apps/server/src/modules/oauth-provider/uc/oauth-provider.client-crud.uc.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@nestjs/common';
+import { OauthProviderService } from '@shared/infra/oauth-provider/index';
+import { OauthClient } from '@shared/infra/oauth-provider/dto/index';
+import { ICurrentUser, Permission, User } from '@shared/domain/index';
+import { AuthorizationService } from '@src/modules/authorization/authorization.service';
+import { UserRepo } from '@shared/repo/index';
+
+@Injectable()
+export class OauthProviderClientCrudUc {
+	constructor(
+		private readonly oauthProviderService: OauthProviderService,
+		private readonly authorizationService: AuthorizationService,
+		private readonly userRepo: UserRepo
+	) {}
+
+	private readonly defaultOauthClientBody: OauthClient = {
+		scope: 'openid offline',
+		grant_types: ['authorization_code', 'refresh_token'],
+		response_types: ['code', 'token', 'id_token'],
+		redirect_uris: [],
+	};
+
+	async listOAuth2Clients(
+		currentUser: ICurrentUser,
+		limit?: number,
+		offset?: number,
+		client_name?: string,
+		owner?: string
+	): Promise<OauthClient[]> {
+		const user: User = await this.userRepo.findById(currentUser.userId);
```

Comment:
you can get the user using `await this.authorizationService.getUserWithPermissions(userId);`
That way, you can avoid having inject the userRepo, and the module doesnt have to include it.

Reply:
good idea. i have changed this